### PR TITLE
[IOTDB-613] Add the ability to persist and maintain currentterm for the raft node

### PR DESCRIPTION
**Pull request by LebronAl**

### CODE REVIEW

File: cluster/src/main/java/org/apache/iotdb/cluster/log/manage/UnCommittedEntryManager.java [hunk 1]
```diff
@@ -126,11 +128,11 @@ public void truncateAndAppend(List<Log> appendingEntries) {
...
         if (len < 0) {
             // the logs are being truncated to before our current offset portion, which is committed entries
-            // unconditional obedience to the leader's request. Maybe throw a exception here is better
-            offset = after;
-            entries = appendingEntries;
+            // Maybe throw a exception here is better
             logger.error("The logs which first index is {} are going to truncate committed logs",
                 after);
+//            throw new TruncateCommittedEntryException(appendingEntries.get(0).getCurrLogIndex(),
+//                offset);
```
Comment (jt2594838) [inline cluster/src/main/java/org/apache/iotdb/cluster/log/manage/UnCommittedEntryManager.java:135]: Please just decide one and do not leave commented code. Both are okay for me.

File: cluster/src/test/java/org/apache/iotdb/cluster/log/manage/RaftLogManagerTest.java [hunk 2]
```diff
@@ -48,12 +52,22 @@ public void apply(Log log) {
...
 	}
 
+	@After
+	public void tearDown() {
+		File dir = new File(new SyncLogDequeSerializer(testIdentifier).getLogDir());
+		for (File file : dir.listFiles()) {
+			file.delete();
+		}
+		dir.delete();
+	}
```
Comment (jt2594838) [inline cluster/src/test/java/org/apache/iotdb/cluster/log/manage/RaftLogManagerTest.java:69]: Please check the indents.

File: cluster/src/test/java/org/apache/iotdb/cluster/log/manage/serializable/SyncLogDequeSerializerTest.java [hunk 3]
```diff
@@ -354,99 +222,87 @@ public void testRemoveOldFileAtRecovery() {
...
+//      List<Log> logs = syncLogDequeSerializer.recoverLog();
+//      assertEquals(9, logs.size());
+//
+//      for (int i = 0; i < 8; i++) {
+//        assertEquals(testLogs2.get(i + 1), logs.get(i));
+//      }
+//    } finally {
+//      syncLogDequeSerializer.close();
+//    }
+//  }
```
Comment (jt2594838) [inline cluster/src/test/java/org/apache/iotdb/cluster/log/manage/serializable/SyncLogDequeSerializerTest.java:286]: Remove these if they are not necessary.

File: cluster/src/test/java/org/apache/iotdb/cluster/log/manage/serializable/SyncLogDequeSerializerTest.java [hunk 4]
```diff
@@ -458,85 +314,78 @@ public void testRecoveryByAppendList() {
...
```
Comment (jt2594838) [inline cluster/src/test/java/org/apache/iotdb/cluster/log/manage/serializable/SyncLogDequeSerializerTest.java:378]: Remove these if they are not necessary.

**Comment by jt2594838** [comment]
Great, only minor issues.

File: cluster/src/main/java/org/apache/iotdb/cluster/log/manage/UnCommittedEntryManager.java (hunk 1 above)
Comment (LebronAl) [reply cluster/src/main/java/org/apache/iotdb/cluster/log/manage/UnCommittedEntryManager.java:135]: OK

File: cluster/src/test/java/org/apache/iotdb/cluster/log/manage/RaftLogManagerTest.java (hunk 2 above)
Comment (LebronAl) [reply cluster/src/test/java/org/apache/iotdb/cluster/log/manage/RaftLogManagerTest.java:69]: It seems wired.In my local editor,the indents are properly.Maybe it ’s a webpage display problem.

File: cluster/src/test/java/org/apache/iotdb/cluster/log/manage/serializable/SyncLogDequeSerializerTest.java (hunk 3 above)
Comment (LebronAl) [reply cluster/src/test/java/org/apache/iotdb/cluster/log/manage/serializable/SyncLogDequeSerializerTest.java:286]: OK

File: cluster/src/test/java/org/apache/iotdb/cluster/log/manage/serializable/SyncLogDequeSerializerTest.java (hunk 4 above)
Comment (LebronAl) [reply cluster/src/test/java/org/apache/iotdb/cluster/log/manage/serializable/SyncLogDequeSerializerTest.java:378]: OK

**Pull request merged**